newlib/libc/tinystdio/conv_flt.c

```diff
@@ -245,7 +245,29 @@ conv_flt (FLT_STREAM *stream, FLT_CONTEXT *context, width_t width, void *addr, u
                 return 0;
 	    }
         }
-	break;
+        if(CHECK_WIDTH() && (i = scanf_getc (stream, context)) == '(') 
+        {
+            while(CHECK_WIDTH() && (i = scanf_getc (stream, context)) != ')') 
+            {
+                if(isdigit(i) || isalpha(i) || i == '_' )
+                {
+                    continue;
+                }
+                else
+                {
+                    break;
+                }
+            }
+            if(i != ')')
+            {
+                    return 0;
+            }
+        }
+        else
+        {
+            scanf_ungetc (i, stream, context);
```

Comment:
Need to avoid this if `CHECK_WIDTH()` fails.

Reply:
Done.